tools/monitoring/cloudmail/mail.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudmail
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	mail "cloud.google.com/go/mail/apiv1alpha3"
+	mailpb "google.golang.org/genproto/googleapis/cloud/mail/v1alpha3"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+const (
+	projectID = "knative-tests"
+	region    = "us-central1"
+
+	addressSetID   = "monitoring-address-set"
+	addressPattern = "monitoring-alert"
+	senderID       = "monitoring-alert-sender"
+	receiptRuleID  = "monitoring-receipt-drop-rule"
+)
+
+// MailClient holds instance of CloudMailClient for making mail-related requests
+type MailClient struct {
+	CloudMailClient *mail.CloudMailClient
+	DomainName      string
+	DomainID        string
+}
+
+// NewMailClient creates a new MailClient to handle all the mail interaction
+func NewMailClient(ctx context.Context) (*MailClient, error) {
+	c, err := mail.NewCloudMailClient(ctx)
+	return &MailClient{
+		CloudMailClient: c,
+		// TODO(yt3liu): Update it with the domain name created in knative-tests
+		DomainName: "",
+		DomainID:   "",
+	}, err
+}
+
+// CreateDomain creates a new project domain for cloud main
+func (c *MailClient) CreateDomain(ctx context.Context) error {
+	domain := &mailpb.Domain{
+		ProjectDomain: true,
+		DomainName:    "",
+	}
+	req := &mailpb.CreateDomainRequest{
+		Parent: fmt.Sprintf("projects/%s", projectID),
+		Region: region,
+		Domain: domain,
+	}
+	resp, err := c.CloudMailClient.CreateDomain(ctx, req)
```

Comment:
We can call `c.CreateDomain()` directly with the change

Reply:
Done